File: nodejs/streaming_pipe_in_and_out.mjs
import * as fs from 'fs';
import * as readline from 'node:readline/promises';

const inputStream = fs.createReadStream('streaming-in.pipe');
const outputStream = fs.createWriteStream('streaming-out.pipe');

const rl = readline.createInterface({ input: inputStream });

for await (const line of rl) {
  outputStream.write(`received ${line}\n`);
}

outputStream.end();

File: python/streaming_pipe_in_and_out.py
input_file = 'streaming-in.pipe'
output_file = 'streaming-out.pipe'

with open(input_file, 'r', encoding='utf-8') as input_pipe:
    with open(output_file, 'w', encoding='utf-8') as output_pipe:
        for line in input_pipe:
            output_pipe.write(f"received {line.strip()}\n")
            output_pipe.flush()
